applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h

```diff
@@ -0,0 +1,1015 @@
+     mBucketSize(ThisParameters["bucket_size"].GetInt()),
+     mSearchFactor(ThisParameters["search_factor"].GetDouble()),
+     mThisInterpolationType(ConvertInter(ThisParameters["interpolation_type"].GetString()))
+     {        
+        if (ThisParameters["internal_variable_interpolation_list"].IsArray() == true)
+        {
+            auto VariableArrayList = ThisParameters["internal_variable_interpolation_list"];
+            
+            for (unsigned int iVar = 0; iVar < VariableArrayList.size(); iVar++)
+            {
+                mInternalVariableList.push_back(KratosComponents<Variable<double>>::Get(VariableArrayList[iVar].GetString()));
+            }
+        }
+        else
+        {
+            std::cout << "WARNING:: No variables to interpolate, look that internal_variable_interpolation_list is correctly defined in your parameters" << std::endl;
+            mInternalVariableList.clear();
+        }
+     }
+    
+    virtual ~InternalVariablesInterpolationProcess(){};
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+    
+    /**
+     * We execute the search relative to the old and new model part
+     */
+    
+    virtual void Execute()
+    {
+        /** NOTE: There are mainly two ways to interpolate the internal variables (there are three, but just two are behave correctly)
+         * CPT: Closest point transfer. It transfer the values from the closest GP
+         * LST: Least-square projection transfer. It transfers from the closest GP from the old mesh
+         * SFT: It transfer GP values to the nodes in the old mesh and then interpolate to the new mesh using the sahpe functions all the time (NOTE: THIS DOESN'T WORK, AND REQUIRES EXTRA STORE)
+         */ 
+        
+        if (mThisInterpolationType == CPT && mInternalVariableList.size() > 0)
+        {
+            InterpolateGaussPointsCPT();
+        }
+        else if (mThisInterpolationType == LST && mInternalVariableList.size() > 0)
+        {                        
+            InterpolateGaussPointsLST();
+        }
+        else if (mThisInterpolationType == SFT && mInternalVariableList.size() > 0)
+        {
+            InterpolateGaussPointsSFT();
+        }
+    }
+
+    /**
+     * This function creates a lists of gauss points ready for the search
+     * @param ThisModelPart: The model part to consider
+     */
+    
+    PointVector CreateGaussPointList(ModelPart& ThisModelPart)
+    {
+        PointVector ThisPointVector;
+        
+        GeometryData::IntegrationMethod ThisIntegrationMethod;
+        
+        // Iterate in the elements
+        ElementsArrayType& pElements = ThisModelPart.Elements();
+        auto numElements = pElements.end() - pElements.begin();
+        
+        const ProcessInfo& CurrentProcessInfo = ThisModelPart.GetProcessInfo();
+
+        #pragma omp for private(ThisIntegrationMethod)
```

Comment:
you can not make this parallel (and actually currently IT IS NOT PARALLEL since you missing the parallel keyword)

the point is that the push_back on line 409 is a serial op